pkg/pillar/dpcreconciler/linux.go

```diff
@@ -1007,6 +1012,53 @@ func (r *LinuxDpcReconciler) rebuildMTUMap(dpc types.DevicePortConfig) {
 	}
 }
 
+func (r *LinuxDpcReconciler) rebuildRouteMetricMap(dpc types.DevicePortConfig) {
+	r.intfMetric = make(map[string]uint32)
+
+	type portInfo struct {
+		logicalLabel string
+		isMgmt       bool
+		cost         uint8
+		index        int
+	}
+
+	var ports []portInfo
+	for i, port := range dpc.Ports {
```

Comment:
The choice of variable names could be more clear. Here "port" is an input and "ports" is an output.
var pi []portInfo 
would improve things.